### PR TITLE
Task: Support Azure Unity Catalog export

### LINKED ISSUE #7551: Unity exporter support for Azure
Error:
```
Error: runtime error: lakefs/catalogexport/unity_exporter.lua:54: external table "prm_maestro_cliente_gt" creation failed: FAILED: BAD_REQUEST [UC_FILE_SCHEME_FOR_TABLE_CREATION_NOT_SUPPORTED] Creating table in Unity Catalog with file scheme https is not supported.
Instead, please create a federated data source connection using the CREATE CONNECTION command for the same table provider, then create a catalog based on the connection with a CREATE FOREIGN CATALOG command to reference the tables therein.
```

Trying to create an external location in databricks using https scheme results in the following error:
```
ErrorClass=INVALID_PARAMETER_VALUE.UNSUPPORTED_LOCATION_SCHEME] url has invalid URI scheme https. Valid URI schemes include s3, gs, s3n, r2, s3a, abfss
```
lakeFS is currently working with https scheme for Azure, and therefore when providing the table path we fail to register the table.

**Note** that according to the above error message external locations are supported only in ADLS gen2 (no `wasb` scheme support)

One solution is to convert the https scheme into an abfss one in the unity exporter before registering the table.

### LINKED ISSUE #7553: Delta exporter: support export of delta log files with abfss scheme to support unity catalog external tables
While working on https://github.com/treeverse/lakeFS/issues/7551 and trying to enable unity exporter on Azure we discovered that in addition to providing the external location as an abfss scheme, we also need to export the delta logs so that the paths are format using this scheme instead of https.

We need to add a flag for the exporter, when using with Azure to decide the format in which to write the paths in the delta log when exporting from lakeFS

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #7551
Closes #7553

## Change Description

Add support for Unity catalog exporter in Azure
Fix delta exporter to support abfss scheme for delta log path and table physical address

### Testing Details

Added esti test case

### Breaking Change?

No

After fix this is the output of delta_exporter:

```

  delta_exporter(completed in 4.949s)

Delta Lake exported table "test-table"'s location: abfss://esti-system-testing@esti4hns.dfs.core.windows.net/cnp12lbck6tb6j1l84lg/testdeltacatalogexportabfss/_lakefs/exported/main/d39916/test_table

Delta Lake exported table "test-table"'s metadata:

	partition_columns = []


	configuration = {}


	created_time = 1707066829815

	description = This is the description of the table

	id = db5e0917-1716-4b0f-a009-c25e5b7304a1

	name = 

	schema_string = {"type":"struct","fields":[{"name":"registration_dttm","type":"timestamp","nullable":true,"metadata":{}},{"name":"id","type":"integer","nullable":true,"metadata":{}},{"name":"first_name","type":"string","nullable":true,"metadata":{}},{"name":"last_name","type":"string","nullable":true,"metadata":{}},{"name":"email","type":"string","nullable":true,"metadata":{}},{"name":"gender","type":"string","nullable":true,"metadata":{}},{"name":"ip_address","type":"string","nullable":true,"metadata":{}},{"name":"cc","type":"string","nullable":true,"metadata":{}},{"name":"country","type":"string","nullable":true,"metadata":{}},{"name":"birthdate","type":"string","nullable":true,"metadata":{}},{"name":"salary","type":"double","nullable":true,"metadata":{}},{"name":"title","type":"string","nullable":true,"metadata":{}},{"name":"comments","type":"string","nullable":true,"metadata":{}},{"name":"__index_level_0__","type":"long","nullable":true,"metadata":{}}]}

```

### CODE REVIEW

File: docs/howto/hooks/lua.md
Comment (Jonathan-Rosenberg) [inline docs/howto/hooks/lua.md:1]: Can you update the `export_delta_log` method with the new parameter?

File: esti/catalog_export_test.go [hunk 1]
```diff
@@ -613,3 +613,101 @@ func getStorageNamespace(t *testing.T, ctx context.Context, repo string) string
...
+	secretAccessKey := viper.GetString("secret_access_key")
+	testData := &exportHooksTestData{
+		Repository:            repo,
+		Branch:                mainBranch,
+		LakeFSAccessKeyID:     accessKeyID,
+		LakeFSSecretAccessKey: secretAccessKey,
+		AzureStorageAccount:   viper.GetString("azure_storage_account"),
+		AzureAccessKey:        viper.GetString("azure_storage_access_key"),
+	}
+	//blockstore := setupCatalogExportTestByStorageType(t, testData)
```
Comment (Jonathan-Rosenberg) [inline esti/catalog_export_test.go:632]: delete?

**Review by Jonathan-Rosenberg — CHANGES_REQUESTED** [review]
Very cool!
Thanks for the quick fix
(just the docs thingy)

File: docs/howto/hooks/lua.md [hunk 2]
```diff
@@ -233,6 +233,11 @@ Sets the object at the given bucket and key to the value of the supplied value s
 Deletes the object at the given key  
 `path_uri` - A valid Azure blob storage uri in the form of `https://myaccount.blob.core.windows.net/mycontainer/myblob`
 
+### `azure/abfss_transform_path(path)`
```
Comment (Isan-Rivkin) [inline docs/howto/hooks/lua.md:236]: 1. Please update [delta_exporter.export_delta_log](https://docs.lakefs.io/howto/hooks/lua.html#lakefscatalogexportdelta_exporterexport_delta_logaction-table_def_names-write_object-delta_client-table_descriptors_path) function doc reference. 

1. assuming most users will use the delta_exporter first and stumble with this issue, please add a bullet in [unity_exporter.register_tables](https://docs.lakefs.io/howto/hooks/lua.html#lakefscatalogexportunity_exporterregister_tablesaction-table_descriptors_path-delta_table_details-databricks_client-warehouse_id) mentioning this.

**Review by Isan-Rivkin — CHANGES_REQUESTED** [review]
Looks good, request only around docs. 
I wonder about friction - not sure so it's more of a question: 
Unity users will have to use this path transformer, maybe we can make it somehow default? 
For example let it be the default in delta_exporter (maybe its a bad idea). 
:)

**Comment by N-o-Z** [comment]
> Looks good, request only around docs. I wonder about friction - not sure so it's more of a question: Unity users will have to use this path transformer, maybe we can make it somehow default? For example let it be the default in delta_exporter (maybe its a bad idea). :)

I don't think making it the default behavior for delta export is a good idea. For delta export use cases this might even be a point of friction for users who have set up a Azure Blob Storage (and not an ADLS gen2 storage account)

File: docs/howto/hooks/lua.md [hunk 3]
```diff
@@ -739,6 +745,10 @@ The registration will use the following paths to register the table:
 `<catalog>.<branch name>.<table_name>` where the branch name will be used as the schema name.
 The return value is a table with mapping of table names to registration request status.
 
+**Note: (Azure users)** Databricks catalog external locations is supported only for ADLS Gen2 storage accounts.  
+When exporting Delta tables using the `lakefs/catalogexport/delta_exporter.export_delta_log` function, the `path_transformer` must be  
+used to convert the paths scheme to `abfss`. the built-in `azure` lua library provides this functionality in `transformPathToAbfss`.
```
Comment (Jonathan-Rosenberg) [inline docs/howto/hooks/lua.md:750]:
```suggestion
used to convert the paths scheme to `abfss`. The built-in `azure` lua library provides this functionality in `transformPathToAbfss`.
```

Just had to...

**Review by Jonathan-Rosenberg — APPROVED** [review]
Awesome (one typo change request that doesn't block)

**Review by Isan-Rivkin — APPROVED** [review]
LGTM!

**Comment by N-o-Z** [comment]
Thanks - waiting to verify the fix completely before merging

**Pull request merged**